application/forms/ChannelForm.php

```diff
@@ -21,10 +30,17 @@ class ChannelForm extends CompatForm
     /** @var ?int Channel ID */
     private $channelId;
 
+/** @var string The current locale */
+    private $locale;
+
     public function __construct(Connection $db, ?int $channelId = null)
     {
         $this->db = $db;
         $this->channelId = $channelId;
+
+        /** @var GettextTranslator $translateInstance */
+        $translateInstance = StaticTranslator::$instance;
+        $this->locale = $translateInstance->getLocale();
     }
 
     protected function assemble()
```

Comment:
please fetch from available_channel_type only once here. And default to the first found type already. There's no need for the user to specifically choose one.